pkg/checks/external/main.go

```diff
@@ -1026,9 +1026,40 @@ func (ext *Checker) createPod() (*apiv1.Pod, error) {
 	// enforce various labels and annotations on all checker pods created
 	ext.addKuberhealthyLabels(p)
 
+	// Get the hostname of the kuberhealthy pod
+	kPodName, err := os.Hostname()
+	if err != nil {
+		ext.log("Unable to get kuberhealthy pod podname")
+	}
+
+	// Get the podspec of kuberhealthy pod
+	kHealthyPod, err := ext.getKuberHealthyPod(kPodName)
+	if err != nil {
+		ext.log("Unable to find kuberhealthy pod", err)
+	} else {
+		// Sets OwnerRefernces on all checker pods
+		p.OwnerReferences = []metav1.OwnerReference{
+			{
+				APIVersion: "v1",
+				Kind:       "Pod",
+				Name:       kPodName,
+				UID:        kHealthyPod.GetUID(),
+			},
+		}
+	}
 	return ext.KubeClient.CoreV1().Pods(ext.Namespace).Create(p)
 }
 
+// getKuberHealthyPod gets the pod spec of kuberhealthy operator
+func (ext *Checker) getKuberHealthyPod(kPodName string) (*apiv1.Pod, error) {
+	podClient := ext.KubeClient.CoreV1().Pods(ext.Namespace)
+	kHealthyPod, err := podClient.Get(kPodName, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+	return kHealthyPod, nil
+}
+
```

Comment:
wondering if this could all be combined to a single function `addOwnerReference` where it takes in the kube client and namespace (like you mentioned with making this more of a global function) and returns an ownerReference.

Reply:
Got an issue with circular imports so i created a separate package that i called utils and made it too two functions where ownerRef calls on the get pod one. Please have a look at the latest commit